frontend/src/metabase/reference/components/FieldTypeDetail.jsx

```diff
@@ -25,25 +23,17 @@ const FieldTypeDetail = ({
       <div className={D.detailTitle}>
         <span>{t`Field type`}</span>
       </div>
-      <div className={cx(D.detailSubtitle, { [CS.mt1]: true })}>
+      <div className={D.detailSubtitle}>
         <span>
           {isEditing ? (
-            <Select
-              placeholder={t`Select a field type`}
+            <SemanticTypePicker
+              field={field}
               value={fieldTypeFormField.value || field.semantic_type}
-              options={FIELD_SEMANTIC_TYPES.concat({
-                id: null,
-                name: t`No field type`,
-                section: t`Other`,
-              }).filter(type =>
-                !isNumericBaseType(field)
-                  ? !(type.id && type.id.startsWith("timestamp_"))
-                  : true,
```

Comment:
No idea what this code was for, but no `type.id` starts with `"timestamp_"` (they all start with `"type/"`), so I just removed it. It should work the same in every context anyway.